### PR TITLE
Nextcloud: fix ldap integration

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/web-apps/nextcloud.nix [hunk 1]
```diff
@@ -551,7 +551,7 @@ in {
             priority = 900;
             extraConfig = if major < 20
               then "rewrite ^ /index.php;"
-              else "try_files $uri $uri/ /index.php$request_uri;";
+              else "try_files $uri/ /index.php$request_uri;";
```
Comment (Mic92) [inline nixos/modules/services/web-apps/nextcloud.nix:554]: This broke autoloading.

Comment (Mic92) [reply nixos/modules/services/web-apps/nextcloud.nix:554]: More info here: https://github.com/nextcloud/documentation/pull/5284

**Comment by ajs124** [comment]
Does this also work on the older, still supported, releases?

**Comment by Ma27** [comment]
@Mic92 I'm not sure if I'm missing something, but it doesn't seem to properly fix the original LDAP issue. I built a VM from your branch (rev c403ed445dbd3cf7f23d86ba0b8b35e8b9f10442) with the following config:

``` nix
# cat vm.nix
{
  snens = { lib, pkgs, config, ... }: {
    networking.firewall.allowedTCPPorts = [ 80 ];
    services.nextcloud = {
      enable = true;
      hostName = "localhost";
      config = {
        adminpass = "root";
      };
    };
  };
}
```

And started it like this:

```
$ nixos-build-vms -I nixpkgs=$(pwd) vm.nix
$ QEMU_NET_OPTS=hostfwd=tcp::8080-:80 ./result/bin/nixos-run-vms
```

When I try to configure LDAP auth in the VM, I still get the following error:

```
snens # [  102.003163] nginx[739]: 2020/10/24 16:58:03 [error] 739#739: *92 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught Error: Class 'OC_JSON' not found in /nix/store/5ybb1r22wxqv92ix92igqp7ysvlyy1wx-nextcloud-20.0.1/apps/user_ldap/ajax/getConfiguration.php:30
snens # [  102.008149] nginx[739]: Stack trace:
snens # [  102.009502] nginx[739]: #0 {main}
snens # [  102.010688] nginx[739]:   thrown in /nix/store/5ybb1r22wxqv92ix92igqp7ysvlyy1wx-nextcloud-20.0.1/apps/user_ldap/ajax/getConfiguration.php on line 30" while reading response header from upstream, client: 10.0.2.2, server: localhost, request: "POST /apps/user_ldap/ajax/getConfiguration.php HTTP/1.1", upstream: "fastcgi://unix:/run/phpfpm/nextcloud.sock:", host: "localhost:8080"
```

**Comment by Mic92** [comment]
> Does this also work on the older, still supported, releases?

It does not change anything for older releases.

**Comment by Mic92** [comment]
I applied now suggestions from https://github.com/nextcloud/documentation/pull/2197#issuecomment-703085112 which is basically a revert of the changes before. Also I think ldap worked with the other hack.

**Comment by Ma27** [comment]
> Also I think ldap worked with the other hack.

@Mic92 which other hack?

**Comment by Mic92** [comment]
> > Also I think ldap worked with the other hack.
> 
> @Mic92 which other hack?

The one you tested in a VM also worked on my server.

**Comment by Ma27** [comment]
Oops sorry, I didn't realize that you pushed new changes after my last commit... LGTM, will merge/backport :)

**Review by Ma27 — APPROVED** [review]

**Comment by Ma27** [comment]
Thanks! Ported to 20.09 as bae0cce2a52e209daa3d4d43ea5025f067dcef0a, 1c50dc407cfc4ed42cf753d2e41499842288d1d8.

**Comment by ajs124** [comment]
Just a note: This leaves behind the `major` function unused.

**Pull request merged**